### DISCUSSION TITLE
Mojo mascot? Python+Mojo=Boitatá (brazilian mythological creature)

### DISCUSSION THREAD

**Discussion by david-ragazzi** [modular/modular#941] [General]
Not a issue itself, but something curious for me being Brazilian which brings me an idea.

Mojo being a superset of Python (or a "Python++"), reminds me of a famous monster in Brazilian folklore: a giant fiery snake that can transform into a ball of fire. His name is Boitatá (of indigenous origin which means "fiery serpent"[1]) or just Bata (name used in other Brazilian regions). The legend is very interesting, watch this video to have an idea about what is this powerful creature:

https://youtu.be/qsLmoINlJL8?si=gwpmcocGv-zI7AVW
 
As Mojo still has not a mascot (Rust has the crab, Python the the snake itself, Linux has the penguin, etc, [2]), how about something which looks like the powerful Boitata? The name itself could be other but I think the image and the legend of Boitata sounds great. See the video to understand what I mean.

Furthermore, this fiery snake mascot would create a connection with Python developers used to see the Python snake symbol. It would sound like a continuation or evolution, not a disruption with Python.

[1] https://mythus.fandom.com/wiki/Boitat%C3%A1
[2] https://en.wikipedia.org/wiki/List_of_computing_mascots

**Comment by alexandreteles** [comment]
> Rust has the *crap*

This made me laugh way more than it should. In any case, I quite like the idea. Although I recognize that the Boitatá isn't well known outside of Brazil, this is an excellent opportunity to adopt a giant snake on fire as a mascot. Maybe someone could doodle what it could look like and present a fascinating character whose lore matches Mojo's "personality" reasonably well.

**Comment by david-ragazzi** [comment]
>This made me laugh way more than it should.

I mean "crab".. :smile: 

> Although I recognize that the Boitatá isn't well known outside of Brazil

Yes, true. I don't understand why Netflix already made 2 seasons of "Invisible City" (series which portrays many characters from Brazilian folklore) and still haven't included Boitata as character. Most critics and specialists are speculating that the 3rd season finally will portray it and so many people outside of Brazil will know it.

>  a fascinating character whose lore matches Mojo's "personality"

yes: fast, powerful, and which protects the system from harmful acts in same way as Boitata.

**Comment by Ronnie5562** [comment]
Hello Mojo lovers,

I am a big fan of Mojo and I have been following it's development. I would like to suggest a mascot for it. I think a brain image with fire on it would be awesome. Here’s why:

A brain image shows the intelligence and creativity of Mojo users. Mojo makes AI development easy and fun for everyone.

A fire image shows the power and speed of Mojo code. Mojo runs on different types of hardware and works with Python libraries.

I made a graphic art of it using Bing chat. I think it looks cool and catchy and we can use something that looks like it.

What do you think? Do you like it? Please share your feedback.
![OIG](https://github.com/modularml/mojo/assets/110787129/a16d69a8-7ef8-418f-810c-f6f7b39f70c0)
![OIG](https://github.com/modularml/mojo/assets/110787129/64bcf839-8344-4ce9-9a92-8bbc4ff4e9fb)

**Reply by rarebreed** [reply]
I personally don't think it's a good idea to try to associate mojicians with a brain/intelligence, as that will inevitably lead to claims by some that "oh those mojicians think they are smarter than everyone else".  Ask the lispers (Smug Lisp Weenies), haskellers (Insufferable Haskell *%cks) and rustaceans (Rust Evangelism Strike Force).

**Comment by david-ragazzi** [comment]
A cute mascot:

![Mojo Boitata](https://i.ytimg.com/vi/6gErX5pNLbU/maxresdefault.jpg)

A not such cute mascot:

![Mojo Boitata](https://s1.static.brasilescola.uol.com.br/be/2021/03/boitata.jpg)

An imponent mascot:

![Mojo Boitata](https://static.wikia.nocookie.net/liberproeliis/images/5/52/Boitata-o-que-e-e-qual-a-origem-da-criatura-nas-lendas-indigenas-960x540.jpg/revision/latest?cb=20220901233059&path-prefix=pt-br)

![Mojo Boitata](https://cdnb.artstation.com/p/assets/images/images/031/787/011/large/guilherme-batista-boitata-forest-guardian-guilherme-batista.jpg?1604588998)

![Mojo Boitata](https://raw.githubusercontent.com/david-ragazzi/mojo/main/Screenshot_20230929_151809.png)

**Reply by fabrizio-ferrari** [reply]
Impressive pictures!

**Comment by fabrizio-ferrari** [comment]
+1 for a big fiery snake as mascot.

**Comment by david-ragazzi** [comment]
Logo ideas:
![Mojo mascot](https://raw.githubusercontent.com/david-ragazzi/mojo/main/Firefly%20A%20giant%20fiery%20python%20wrapped%20in%20the%20word%20-Mojo-%2042261.jpg)

![Mojo mascot](https://raw.githubusercontent.com/david-ragazzi/mojo/main/Firefly%20A%20logo%20with%20a%20giant%20snake%20on%20fire%2076406.jpg)

![Mojo mascot](https://raw.githubusercontent.com/david-ragazzi/mojo/main/Firefly%20A%20logo%20with%20a%20cute%20snake%20on%20fire%2088007.jpg)

![Mojo mascot](https://raw.githubusercontent.com/david-ragazzi/mojo/main/Firefly%20A%20cute%20snake%20on%20fire%2077315.jpg)

![Mojo mascot](https://raw.githubusercontent.com/david-ragazzi/mojo/main/Firefly%20A%20cute%20snake%20on%20fire%2075777.jpg)

![Mojo mascot](https://raw.githubusercontent.com/david-ragazzi/mojo/main/Firefly%20A%20cute%20snake%20on%20fire%2062240.jpg)

**Comment by rarebreed** [comment]
I like this.  The serpent also goes hand in hand with python.  And that Boitatá looks more powerful than a python ;)

**Comment by rarebreed** [comment]
I have zero computer art skills, but a magician pulling a python out of a hat would be kind of catchy :)

**Comment by edilalves** [comment]
+1 

I liked this one as mascot.

**Comment by ivellapillil** [comment]
How about the python shaped like M with fire in mouth?

![image](https://github.com/modularml/mojo/assets/29715511/40a106f6-b52a-4ee1-8a37-540d2254e943)
![image](https://github.com/modularml/mojo/assets/29715511/bc331d86-c717-4504-825e-b8d012c13877)
![image](https://github.com/modularml/mojo/assets/29715511/64b099e0-c975-4ffe-a12e-2da708491b4c)

**Reply by david-ragazzi** [reply]
> How about the python shaped like M with fire in mouth?

I like the idea and actually I tried it. But honestly, I don't think FireFly is doing a good job when we ask a snake with a M letter. My own experience: I tried things like "a giant snake on fire wrapped in a M letter". My only results not so bad it was when I typed "a giant snake on fire BEHIND a M letter". Even so they're not such good as the snake alone. The AI algorithm try fill the M letter with a snake in a very bizarre way mainly the texture. By the way, my FireFly credits were gone to test with more variations.

**Reply by ivellapillil** [reply]
Same experience here. Tried 1 hour using various engines. None could do the python in shape of M. I am not good at illustrations, so pasted graphics that was at least good enough 😅

**Comment by david-ragazzi** [comment]
Regarding the mascot's name, all mascot names are short like "Tux" for the Linux penguin, "Ferris" for the Rust crab, and so on. I think "Bata" or "Tata" [1] would be a nice and appropriate nickname for a mascot based on Boitatá. Both of them are short and easy to remember names, and maintains the connection with the origin of the mythical character. 

Other names which Boitata also is called in Brazil and could have nick variations from them:
Baitatá
Batatá
Boi-íra
Botoitá
Bitatá
M'boi-tatá

[1] "Tata" means "fire" in indigenous Guarani language which something very familiar to us automatically comes to mind: :fire:

**Comment by duartemolha** [comment]
The mascot should definitely be called monthy

**Reply by Sharktheone** [reply]
Then we only need to form Monty Mojo 😉

**Comment by gryznar** [comment]


**Reply by david-ragazzi** [reply]
IMHO, I didn't like it. The reasons:
- No immediate recognition or self identification from Python developers (as myself). Sounds like a rupture instead of a continuation or evolution of Python.
- To make logos, illustrations, etc, will be harder because they're are 2 creatures into one. Beyond the fact, there're no 2-mascot-into-1 approach anywhere. Every mascot is a single creature...

**Reply by gryznar** [reply]
I agree. Python's root are missing. But this flame and whole symbolism looks really cool! So maybe not a mascot, but something else?

Edit: deleted. You are right
